Actions/PSScriptAnalyzer/action.yml

```diff
@@ -0,0 +1,62 @@
+name: Run PSScriptAnalyzer
+author: Microsoft Corporation
+branding:
+  icon: "check"
+  color: "gray-dark"
+inputs:
+  path:
+    description: 'Specifies the path to the scripts or module to be analyzed. Wildcard characters are supported.'
+    required: true
+    default: '.\'
+  excludeRule:
+    description: 'Comma separated list of PSScriptAnalyzer rules to exclude. Wildcard characters are supported.'
+    required: false
+  recurse:
+    description: 'Runs Script Analyzer on the files in the Path directory and all subdirectories recursively.'
+    required: false
+  output:
+    description: 'Specifies where the path for the sarif file'
+    required: true
+    default: 'results.sarif'
+runs:
+  using: "composite"
+  steps:
+    - name: Run PSScriptAnalyzer and ConvertToSARIF
+      shell: pwsh
+      run: |
+        $analyzerModule = Get-Module -ListAvailable -Name PSScriptAnalyzer
```

Comment:
Consider extracting the script in a .ps1 file. Powershell in YAML isn't as readable and, more importantly, isn't getting analyzed.